core/sessions/ldapauth/ldap.go

```diff
@@ -0,0 +1,861 @@
+package ldapauth
+
+/*
+
+The LDAP authentication module forwards the credentials in the user session request
+for authentication with a configured upstream LDAP server
+
+This module relies on the two following local database tables:
+	ldap_sessions: 	Upon successful LDAP response, creates a keyed local copy of the user
+					email
+	ldap_user_api_tokens: User created API tokens, tied to the node, storing user email.
+						  Note: user can have only one API token at a time, and token
+						  expiration is enforced
+
+User session and roles are cached and revalidated with the upstream service at the interval defined in
+the local LDAP config through the Application.sessionReaper implementation in reaper.go.
+Changes to the upstream identity server will propagate through and update local tables (web sessions, API tokens)
+by either removing the entries or updating the roles. This sync happens for every auth endpoint hit, and
+via the defined sync interval. One goroutine is created to coordinate the sync timing in the New function
+
+This implementation is read only; user mutation actions such as Delete are not supported.
+
+MFA is supported via the remote LDAP server implementation. Sufficient request time out should accommodate
+for a blocking auth call while the user responds to a potential push notification callback.
+
+*/
```

Comment:
This comment is unattached and so gets dropped during godoc generation. It will render as a package comment if you attach it to the package name like this:
```suggestion
/*

The LDAP authentication module forwards the credentials in the user session request
for authentication with a configured upstream LDAP server

This module relies on the two following local database tables:
	ldap_sessions: 	Upon successful LDAP response, creates a keyed local copy of the user
					email
	ldap_user_api_tokens: User created API tokens, tied to the node, storing user email.
						  Note: user can have only one API token at a time, and token
						  expiration is enforced

User session and roles are cached and revalidated with the upstream service at the interval defined in
the local LDAP config through the Application.sessionReaper implementation in reaper.go.
Changes to the upstream identity server will propagate through and update local tables (web sessions, API tokens)
by either removing the entries or updating the roles. This sync happens for every auth endpoint hit, and
via the defined sync interval. One goroutine is created to coordinate the sync timing in the New function

This implementation is read only; user mutation actions such as Delete are not supported.

MFA is supported via the remote LDAP server implementation. Sufficient request time out should accommodate
for a blocking auth call while the user responds to a potential push notification callback.

*/
package ldapauth
```
![image](https://github.com/smartcontractkit/chainlink/assets/1194128/2f4e8afc-769e-4ba5-b679-853c27cd6265)
You can run `godoc -http=:6060` to test locally

Reply:
Oh that's awesome! Updated now thanks for the tip